cpp/oneapi/dal/algo/kmeans/backend/cpu/train_kernel_lloyd_dense.cpp

```diff
@@ -83,63 +83,106 @@ static daal::data_management::NumericTablePtr get_initial_centroids(
     return daal_initial_centroids;
 }
 
+inline auto get_daal_parameter_to_train(const descriptor_t& desc) {
+    const std::int64_t cluster_count = desc.get_cluster_count();
+    const std::int64_t max_iteration_count = desc.get_max_iteration_count();
+    const double accuracy_threshold = desc.get_accuracy_threshold();
+
+    daal_kmeans::Parameter par(dal::detail::integral_cast<std::size_t>(cluster_count),
+                               dal::detail::integral_cast<std::size_t>(max_iteration_count));
+
+    par.accuracyThreshold = accuracy_threshold;
+
+    const auto res_op = desc.get_result_options();
+    bool all_metrics = res_op.test(result_options::compute_centroids) &&
+                       res_op.test(result_options::compute_exact_objective_function) &&
+                       res_op.test(result_options::compute_assignments);
+
+    if (!all_metrics) {
+        par.resultsToEvaluate = static_cast<DAAL_UINT64>(daal_kmeans::computeCentroids);
```

Comment:
What if !all_metrics but result_options::compute_assignments is true?

Reply:
good point, I will fix it. The current fixes mostly targeted to scikit-learn-intelex

Reply:
I think that in general we need to support 2 scenarios in train(compute only centroids / compute everything), because for compute just obj func or assignments we can use infer, so, you can check my changes in the latest commit

Reply:
Sounds good